### PR TITLE
feat: add kubeconfig helpers for k8s

### PR DESCRIPTION
**Pull request by Sh4d1**
Signed-off-by: Patrik Cyvoct <pcyvoct@scaleway.com>

### CODE REVIEW

File: api/k8s/v1beta3/kubeconfig.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		Region:    req.Region,
+		ClusterID: req.ClusterID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	kubeconfigContent, err := ioutil.ReadAll(kubeconfigFile.Content)
+	if err != nil {
+		return nil, err
```
Comment (kindermoumoute) [inline api/k8s/v1beta3/kubeconfig.go]: should return a wrapped error `errors.Wrap()`

File: api/k8s/v1beta3/kubeconfig.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	}
+
+	var kubeconfig Kubeconfig
+	err = yaml.Unmarshal(kubeconfigContent, &kubeconfig)
+	if err != nil {
+		return nil, err
```
Comment (kindermoumoute) [inline api/k8s/v1beta3/kubeconfig.go]: should return a wrapped error `errors.Wrap()`

File: api/k8s/v1beta3/kubeconfig.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import (
+	"io/ioutil"
+
+	"github.com/scaleway/scaleway-sdk-go/internal/errors"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	"gopkg.in/yaml.v2"
+)
+
+// Kubeconfig represents a kubernetes kubeconfig file
+type Kubeconfig struct {
```
Comment (QuentinBrosse) [inline api/k8s/v1beta3/kubeconfig.go]: Add a `raw []byte` field containing the raw config.

Comment (QuentinBrosse) [inline api/k8s/v1beta3/kubeconfig.go]: Declare types for nested structs.

File: api/k8s/v1beta3/kubeconfig.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	var kubeconfig Kubeconfig
+	err = yaml.Unmarshal(kubeconfigContent, &kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
```
Comment (QuentinBrosse) [inline api/k8s/v1beta3/kubeconfig.go]: Maybe you can check the number of `clusters` and `users` here and not in the getters.
More concrete error messages could be great too. Something like: `the kubeconfig file must contain only one user`.

**Pull request merged**